### ISSUE TITLE
Create new ``no-self-class-in-super`` message

### ISSUE THREAD

**Issue by DanielNoord** [pylint-dev/pylint#6961]
### Current problem

Ref. https://github.com/PyCQA/pylint/issues/4922

We will no longer emit `bad-super-call` for:

```python
class HasWhoAmI:
    def whoami(self):
        pass

class ClassB(HasWhoAmI):
    def whoami(self):
        super().whoami()

class ClassC(HasWhoAmI):
    def whoami(self):
        super().whoami()

class ClassD(ClassB, ClassC):
    def greet(self):
        super(ClassB, self).whoami()  # previously raised
```

However, it is often best practice to use `ClassD` instead of `ClassB` (so the `self-class`) in a `super()` call.

### Desired solution

Raise ``no-self-class-in-super``.

I'm not sure if that's the best name though and at which level this message should be.

@NeilGirdhar Do you have thoughts about this?

**Comment by Pierre-Sassoulas** [comment]
Regarding the message name how about ``super-called-with-parent`` or ``parent-class-in-super`` ?

**Comment by DanielNoord** [comment]
That wouldn't cover:
```python
class A:
    def method(self):
        ...

class B:
    def method(self):
        super(A, self).method()
```

That would be covered by `bad-super-call` but normally we try to avoid dependency between messages.

**Comment by Pierre-Sassoulas** [comment]
But if we raise the new message for this example, it means we would be raising a convention or a refactor message on a piece of code that would crash if run. This is the reason why we have some message grouped in checkers, so we can have consistency easily between message that are dependent on each other. One case is already handled by ``bad-super-call`` we don't have to raise twice. For example we don't raise ``useless-option-value`` each time we raise an ``unknown-error-value``.

**Comment by DanielNoord** [comment]
> But if we raise the new message for this example, it means we would be raising a convention or a refactor message on a piece of code that would crash if run. This is the reason why we have some message grouped in checkers, so we can have consistency easily between message that are dependent on each other. One case is already handled by `bad-super-call` we don't have to raise twice. For example we don't raise `useless-option-value` each time we raise an `unknown-error-value`.

I don't think it avoidable to raise the message here without making the checker-logic very difficult. As I imagine it know the logic is quite simple:
```python
if call is supercall: # ...
    if call.args[0] is not functiondef.frame():
        self.add_message(...)
```

Checking if `args[0]` is in the `mro()` is unnecessary I think.

**Comment by jacobtylerwalls** [comment]
This is essentially the second sub-proposal in #6992 (`order-dependent-super-resolution`)

**Comment by jacobtylerwalls** [comment]
Unlinking #7007, since no longer has to do with `super`.
